nni/common/concrete_trace_utils/operator_patcher.py

```diff
@@ -226,20 +229,27 @@ def patch_inner_helper(self, func):
                 assert _orig_len(closures) == _orig_len(co_freevars)
                 closure_dict = _orig_dict(_orig_zip(co_freevars, [c.cell_contents for c in closures]))
 
-            var_dict = {}
-            exec(
-                # use func.__code__.co_filename to make the new function easily debuggable.
-                compile(new_tree, func_inner.__code__.co_filename, 'exec'),
-                {
-                    'patch_run': OperatorPatcherContext.patch_run,
-                    **func_inner.__globals__,
-                    **closure_dict,
-                },
-                var_dict)
-            if the_self is not None:
-                return var_dict['new_func'].__get__(the_self)
-            else:
-                return var_dict['new_func']
+            tuple_wrapped = tuple
+            try:
+                if platform.python_version_tuple() < ('3', '9'):
+                    setattr(builtins, 'tuple', _orig_tuple)
+                var_dict = {}
+                exec(
+                    # use func.__code__.co_filename to make the new function easily debuggable.
+                    compile(new_tree, func_inner.__code__.co_filename, 'exec'),
+                    {
+                        'patch_run': OperatorPatcherContext.patch_run,
+                        **func_inner.__globals__,
+                        **closure_dict,
+                    },
+                    var_dict)
+                if the_self is not None:
+                    return var_dict['new_func'].__get__(the_self)
+                else:
+                    return var_dict['new_func']
+            finally:
+                if platform.python_version_tuple() < ('3', '9'):
+                    setattr(builtins, 'tuple', tuple_wrapped)
```

Comment:
what are these for?

Reply:
the type hint `typing.Tuple` will meet error if the tuple is wrapped under `python < 3.9`